plugins/aws_sts/app/decorators/stage_decorator.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+Stage.class_eval do
+  validates(
+    :aws_sts_iam_role_arn,
+    allow_blank: true,
+    format: {
+      with: /\Aarn:aws:iam::\d+:role\/.+\Z/,
+      message: "does not have the correct syntax. Must be of the form: arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME"
+    }
+  )
+
+  validates(
+    :aws_sts_iam_role_session_duration,
+    allow_blank: true,
+    numericality: {
+      only_integer: true,
+      greater_than_or_equal_to: SamsonAwsSts::SESSION_DURATION_MIN,
+      less_than_or_equal_to:    SamsonAwsSts::SESSION_DURATION_MAX
+    }
+  )
+
+  validate :can_assume_role, if: -> { self&.aws_sts_iam_role_arn.present? }
+
+  before_save do
```

Comment:
prefer named callbacks so they are easy to see in the backtrace and stubable in tests